src/server/api/controllers/orders.controller.js

```diff
@@ -0,0 +1,40 @@
+const knex = require('../../config/db');
+const HttpError = require('../lib/utils/http-error');
+
+const getOrdersByUserId = async (raw_user_id) => {
+  const user_id = parseInt(raw_user_id, 10);
+
+  if (isNaN(user_id) || user_id < 1) {
+    throw new HttpError('The ID should be a number', 400);
+  }
+  try {
+    const orders = await knex('orders')
+      .select(
+        'orders.user_id',
+        'orders.id as order_number',
+        'orders.updated_at',
+        'orders.status',
+        knex.raw('sum( ?? * ?? ) as ??', ['quantity', 'price', 'total_amount']),
+      )
+      .leftJoin('order_items', 'order_items.order_id', '=', 'orders.id')
+      .leftJoin('users', 'users.id', '=', 'orders.user_id')
+      .leftJoin('products', 'products.id', '=', 'order_items.product_id')
+      .where({ user_id })
+      .groupBy('orders.id')
+      .count('orders.id as nr_of_items');
+    if (orders.length === 0) {
+      throw new HttpError('The user ID you provided does not exist', 404);
+    }
+
+    return new HttpError('The user have no orders yet', 404);
```

Comment:
```suggestion
    if (orders.length === 0) {
      throw new HttpError('The user have no orders yet', 404);
    }

    return orders;
```

This seems like a little edit mistake, let's fix it and merge :)

Reply:
Check updated one please😌